src/gluonts/mx/model/deepvar_hierarchical/_estimator.py

```diff
@@ -82,14 +85,29 @@ def null_space_projection_mat(A: np.ndarray) -> np.ndarray:
     A
         The constraint matrix A in the equation: Ay = 0 (y being the
         values/forecasts of all time series in the hierarchy).
+    D
+        Positive definite matrix (typically a diagonal matrix). Optional.
+        If provided then the distance between the reconciled and unreconciled
+        forecasts is calculated based on the norm induced by D. Useful for
+        weighing the distances differently for each level of the hierarchy.
+        By default Euclidean distance is used.
 
     Returns
     -------
     Numpy ND array
         Projection matrix, shape (total_num_time_series, total_num_time_series)
     """
     num_ts = A.shape[1]
-    return np.eye(num_ts) - A.T @ np.linalg.pinv(A @ A.T) @ A
+    if D is None:
+        return np.eye(num_ts) - A.T @ np.linalg.pinv(A @ A.T) @ A
+    else:
+        assert np.all(
+            np.linalg.eigvals(D) > 0
+        ), "`D` must be positive definite."
+        D_inv = np.linalg.inv(D)
```

Comment:
This is a good assert check. Can we add an assert to verify that the matrix is symmetric?
If I understand correctly, the proposed oblique projection matrix is derived from the induced norm $\|\|x\|\|^2_A=x^TAx$. 
Using a check on symmetry before the check on eigenvalues would be good as the former is cheaper than the latter.